js/modules/backup.js

```diff
@@ -810,11 +819,17 @@ function saveMessage(db, message) {
   return saveAllMessages(db, [message]);
 }
 
-function saveAllMessages(db, messages) {
-  if (!messages.length) {
+async function saveAllMessages(db, rawMessages) {
+  if (rawMessages.length === 0) {
     return Promise.resolve();
   }
 
+  const { importMessage, upgradeMessageSchema } = Signal.Migrations;
+  const importAndUpgrade = async message =>
+    upgradeMessageSchema(await importMessage(message));
```

Comment:
`importMessage` preserve the existing shape and simply stores disk attachments in the right place. After that, the upgrade pipeline kicks in and takes care of upgrading the messages to the latest schema version.

Reply:
Thanks. This might be a good comment for the code itself.

Reply:
In fact, something like `writeAttachmentToDiskIfNeeded` would be more explanatory name here, without the need for the comment.

Reply:
I like it! Always prefer to change code instead of comment if possible as it doesn’t go out of date. Now I just gotta figure out how to make that name with `create*` prefix and fitting for SSDs as well 😉